docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md

```diff
@@ -19,85 +19,83 @@ weight: 1000
+
+1. Optional: [Configure a refresh token]({{< relref "#configure-a-refresh-token" >}}):
 
-You may have to set the `root_url` option of `[server]` for the callback URL to be
-correct. For example in case you are serving Grafana behind a proxy.
+   a. Enable `accessTokenExpirationCheck` feature toggle.
 
-Restart the Grafana backend for your changes to take effect.
+   c. Set `use_refresh_token` to `true` in `[auth.generic_oauth]` section in Grafana configuration file.
 
-With `allow_sign_up` set to `false`, only existing users will be able to login
-using their GitLab account, but with `allow_sign_up` set to `true`, _any_ user
-who can authenticate on GitLab will be able to login on your Grafana instance;
-if you use the public `gitlab.com`, it means anyone in the world would be able
-to login on your Grafana instance.
+1. [Configure role mapping]({{< relref "#configure-role-mapping" >}}).
+1. Optional: [Configure team synchronization]({{< relref "#configure-team-synchronization" >}}).
+1. Restart Grafana.
 
-You can limit access to only members of a given group or list of
-groups by setting the `allowed_groups` option.
+   You should now see a GitLab login button on the login page and be able to log in or sign up with your GitLab accounts.
 
-You can also specify the SSL/TLS configuration used by the client.
+## Configuration options
 
-- Set `tls_client_cert` to the path of the certificate.
-- Set `tls_client_key` to the path containing the key.
-- Set `tls_client_ca` to the path containing a trusted certificate authority list.
+The table below describes all GitLab OAuth configuration options. Like any other Grafana configuration, you can apply these options as environment variables.
 
-`tls_skip_verify_insecure` controls whether a client verifies the server's certificate chain and host name. If it is true, then SSL/TLS accepts any certificate presented by the server and any host name in that certificate. _You should only use this for testing_, because this mode leaves SSL/TLS susceptible to man-in-the-middle attacks.
+| Setting                      | Required | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Default                              |
```

Comment:
nit/not needed for merge/seeking opinion: Is the order we present these options the best?

Should it be required first? 
Isn't enabled required for pratical purposes :D?

Reply:
Good point, good point :) I'll reorder them to have the required options first, and also set `enabled` to required.